thespian/test/test_wakeup.py

```diff
@@ -27,15 +27,22 @@
 
 class RetryActor(Actor):
     def __init__(self):
-        self._numWakeups = 0
+        self._numWakeups = {}
     def receiveMessage(self, msg, sender):
-        if "check" == msg:
-            self.wakeupAfter(wakeupAfterPeriod)
+        if "check" == msg or "awoken?" == msg:
+            self.dispatch(sender, (msg, None))
         elif isinstance(msg, WakeupMessage):
-            self._numWakeups += 1
-        elif "awoken?" == msg:
-            self.send(sender, self._numWakeups)
-
+            if msg.payload not in self._numWakeups:
+                self._numWakeups[msg.payload] = 0
+            self._numWakeups[msg.payload] += 1
```

Comment:
FYI, this can be simplified using `defaultdict`.  In line 30 above, use`self._numWakeups = defaultdict(int)` and then you can remove lines 35 and 36.  

The code here is fine, especially given that this is a test, but `defaultdict` is an often overlooked gem for these types of situations, so I thought I'd mention it in case you weren't aware of it.